sympy/series/sequences.py

```diff
@@ -711,6 +714,154 @@ def coeff_mul(self, coeff):
         return SeqFormula(formula, self.args[1])
 
 
+class RecursiveSeq(SeqBase):
+    """A finite degree recursive sequence.
+
+    That is, a sequence a(n) that depends on a fixed, finite number of its
+    previous values. The general form is
+
+        a(n) = f(a(n - 1), a(n - 2), ..., a(n - d))
+
+    for some fixed, positive integer d, where f is some function defined by a
+    SymPy expression.
+
+    Parameters
+    ==========
+
+    recurrence : SymPy expression defining recurrence
+        This is *not* an equality, only the expression that the nth term is
+        equal to. For example, if :code:`a(n) = f(a(n - 1), ..., a(n - d))`,
+        then the expression should be :code:`f(a(n - 1), ..., a(n - d))`.
+
+    y : function with argument
+        The name of the recursively defined sequence, e.g., :code:`y(n)`.
+
+    initial : iterable with length equal to the degree of the recurrence
+        The initial values of the recurrence.
+
+    start : start value of sequence (inclusive)
+
+    stop : stop value of sequence (inclusive)
+
+    Examples
+    ========
+
+    >>> from sympy import Function, symbols
+    >>> from sympy.series.sequences import RecursiveSeq
+    >>> y = Function("y")
+    >>> n = symbols("n")
+    >>> fib = RecursiveSeq(y(n - 1) + y(n - 2), y(n), [0, 1])
+    >>> fib.coeff(3) # Value at a particular point
+    2
+    >>> fib[:6] # supports slicing
+    [0, 1, 1, 2, 3, 5]
+    >>> fib.recurrence # inspect recurrence
+    Eq(y(n), y(n - 2) + y(n - 1))
+    >>> fib.degree # automatically determine degree
+    2
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqFormula
+
+    """
+
+    def __new__(cls, recurrence, y, initial=None, start=0, stop=oo):
```

Comment:
How about replacing `y` with hidden argument by explicit `y` and `n`? Perhaps it could also be possible to do without `stop`.

Reply:
I was following the example in `rsolve`, but I agree that the hidden argument is annoying. Would it be preferable to split the two up even if it conflicts with how `rsolve` works?

Reply:
I have no strong preference either way. In any case, `n` should probably be checked to avoid arguments like `y(3)`.

Reply:
I'll look into which would be better, but check `n.is_symbol` either way.

Also, to the `stop` point, I'm worried about recurrences like `y(n) = y(n - 1) / (n - 6)`. This is defined only for, say, `0 <= n < 6`.

Reply:
I think that it is up to the user to check the return value for infinities or undefined values.